src/node/history.h

```diff
@@ -55,7 +55,11 @@ namespace ccf
     COMPACT
   };
 
+#ifdef OVERRIDE_MAX_HISTORY_LEN
+  constexpr int MAX_HISTORY_LEN = OVERRIDE_MAX_HISTORY_LEN;
+#else
   constexpr int MAX_HISTORY_LEN = 1000;
+#endif
```

Comment:
This is now overridden by the `historical_queries` unit test. The problem is that with the default value, when we only create a few dozen entries, _any_ signature contains a minitree that covers every single transaction. This gets in the way of testing that we're always finding a signature! Note that this history length actually introduces a really nasty subtlety in the historical queries - the receipt we return _may not be the canonically shortest receipt_ that you'd get previously, instead it's just _a_ receipt from a future signature transaction, but if we're unlucky (in the order we receive ledge entries) it may be 3 signatures in the future rather than the _next_ signature. I think we can avoid that if we're careful, but haven't confirmed that since I'm not sure if the canonical receipt concept is actually necessary.

Reply:
Until we've worked out if canonical receipts are particularly interesting somehow, this is a good comment to add but not an actual issue.